node-guidelines/checklist.md

```diff
@@ -0,0 +1,29 @@
+## Checklist
+Use this checklist to help you when developing an application
+
+### Application setup
+- [ ] **Using node v10**
+- [ ] **Using typescript**
+- [ ] **tsconfig and tslint are set up in standard way**
+- [ ] **Using express**
+
+### Build
+- [ ] **Using pipeline provided**
+
+### Development
+- [ ] **lower-case filenames**
```

Comment:
[ ] implement REST routes in separate file
[ ] implement REST API help as HTML with links to available url endpoints.

Reply:
We use swagger for our Java API documentation. It would be nice if our API docs were the same regardless of the language used to implement the service.

Unfortunately, the primary [swagger package for node](https://github.com/swagger-api/swagger-node) appears to be opinionated about project structure, but the [underlying middleware](https://github.com/apigee-127/swagger-tools/blob/master/docs/Middleware.md) may be usable.

The less popular [typescript-rest-swagger](https://github.com/thiagobustamante/typescript-rest-swagger) may be worth a look, as well.

Reply:
I'll create an issue for this

Reply:
@AustinMatherne @ThomasBurleson #8